packages/edit-site/src/components/page-patterns/rename-category-menu-item.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * WordPress dependencies
+ */
+import { MenuItem } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { privateApis as patternsPrivateApis } from '@wordpress/patterns';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+
+const { RenamePatternCategoryModal } = unlock( patternsPrivateApis );
+
+export default function RenameCategoryMenuItem( { category, onClose } ) {
+	const [ isModalOpen, setIsModalOpen ] = useState( false );
+
+	if ( ! category?.id ) {
+		return null;
+	}
+
+	return (
+		<>
+			<MenuItem onClick={ () => setIsModalOpen( true ) }>
+				{ __( 'Rename' ) }
+			</MenuItem>
+			{ isModalOpen && (
+				<RenamePatternCategoryModal
+					categoryId={ category.id }
```

Comment:
Maybe we can simply pass `category` here so that we can save us an additional network request.

Reply:
My initial approach was trying to pass the category object through to the rename modal however I was running into an error when attempting to save the edited entity record. 

<img width="375" alt="Screenshot 2023-10-09 at 12 51 47 pm" src="https://github.com/WordPress/gutenberg/assets/60436221/ca118af5-4ef9-46b1-b154-342e27ba0226">


As I understand it, given the user pattern categories were retrieved as a collection, there isn't an [individual entity record object](https://github.com/WordPress/gutenberg/blob/b244819d180c9f14d20984cad86be5b1aeb24d3b/packages/core-data/src/actions.js#L362) available so the `editEntityRecord` call fails. 

There's very likely something I'm missing, so I'm open to alternative approaches here 🙏 

<details>
<summary>Example diff passing the category object rather than ID</summary>

```diff
diff --git a/packages/edit-site/src/components/page-patterns/rename-category-menu-item.js b/packages/edit-site/src/components/page-patterns/rename-category-menu-item.js
index fcb46a09fb..fe1e1b5587 100644
--- a/packages/edit-site/src/components/page-patterns/rename-category-menu-item.js
+++ b/packages/edit-site/src/components/page-patterns/rename-category-menu-item.js
@@ -27,7 +27,7 @@ export default function RenameCategoryMenuItem( { category, onClose } ) {
 			</MenuItem>
 			{ isModalOpen && (
 				<RenamePatternCategoryModal
-					categoryId={ category.id }
+					category={ category }
 					onClose={ () => {
 						setIsModalOpen( false );
 						onClose();
diff --git a/packages/patterns/src/components/rename-pattern-category-modal.js b/packages/patterns/src/components/rename-pattern-category-modal.js
index 03280910d6..753069a3ea 100644
--- a/packages/patterns/src/components/rename-pattern-category-modal.js
+++ b/packages/patterns/src/components/rename-pattern-category-modal.js
@@ -9,8 +9,8 @@ import {
 	__experimentalVStack as VStack,
 } from '@wordpress/components';
 import { store as coreStore } from '@wordpress/core-data';
-import { useDispatch, useSelect } from '@wordpress/data';
-import { useEffect, useState } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+import { useState } from '@wordpress/element';
 import { decodeEntities } from '@wordpress/html-entities';
 import { __ } from '@wordpress/i18n';
 import { store as noticesStore } from '@wordpress/notices';
@@ -20,39 +20,10 @@ import { store as noticesStore } from '@wordpress/notices';
  */
 import { CATEGORY_SLUG } from './category-selector';
 
-const usePatternCategory = ( categoryId ) => {
-	return useSelect(
-		( select ) => {
-			const record = select( coreStore ).getEditedEntityRecord(
-				'taxonomy',
-				CATEGORY_SLUG,
-				categoryId
-			);
-			const hasResolvedRecord = select( coreStore ).hasFinishedResolution(
-				'getEditedEntityRecord',
-				[ 'taxonomy', CATEGORY_SLUG, categoryId ]
-			);
-
-			return {
-				category: record,
-				hasResolved: hasResolvedRecord,
-			};
-		},
-		[ categoryId ]
-	);
-};
-
-export default function RenamePatternCategoryModal( { categoryId, onClose } ) {
-	const { category, hasResolved } = usePatternCategory( categoryId );
-	const [ name, setName ] = useState( '' );
+export default function RenamePatternCategoryModal( { category, onClose } ) {
+	const [ name, setName ] = useState( decodeEntities( category.label ) );
 	const [ isSaving, setIsSaving ] = useState( false );
 
-	useEffect( () => {
-		if ( hasResolved && category?.name ) {
-			setName( decodeEntities( category.name ) );
-		}
-	}, [ hasResolved, category?.name ] );
-
 	const {
 		editEntityRecord,
 		invalidateResolution,
@@ -68,10 +39,11 @@ export default function RenamePatternCategoryModal( { categoryId, onClose } ) {
 		if ( ! name || name === category.name || isSaving ) {
 			return;
 		}
+
 		try {
 			setIsSaving( true );
 
-			await editEntityRecord( 'taxonomy', CATEGORY_SLUG, categoryId, {
+			await editEntityRecord( 'taxonomy', CATEGORY_SLUG, category.id, {
 				name,
 			} );
 
@@ -81,7 +53,7 @@ export default function RenamePatternCategoryModal( { categoryId, onClose } ) {
 			await saveSpecifiedEntityEdits(
 				'taxonomy',
 				CATEGORY_SLUG,
-				categoryId,
+				category.id,
 				[ 'name' ],
 				{ throwOnError: true }
 			);

```

</details>

Reply:
What if we simply call `saveEntityRecord` instead of `editEntityRecord` then `saveSpecifiedEntityEdits`?

Reply:
🤦 I had blinkers on after looking at recent edits that had merged templates, template parts, and patterns actions. For some reason, I was thinking I needed to avoid saving other edits still.

Thanks for the suggestion, `saveEntityRecord` works fine. 

My only questions now are:
1.  Should this modal only have to expect a category object that matches the entity?
    - User-created pattern categories retrieved via `getUserPatternCategories` fudge the category object's properties.
    - Adding a label prop assigning the name value to that
    - Overriding the name prop setting that to the slug value
2. If we are renaming the pattern, should it also be saving a new slug?

Reply:
> Should this modal only have to expect a category object that matches the entity?

It doesn't seem quite right that the patterns package should have intimate knowledge of how the other packages might manipulate the categories. I would be inclined to expect a standard category entity in the  `rename-pattern-category-modal.js` and leave it to the site editor to pass this in correctly.

> If we are renaming the pattern, should it also be saving a new slug?

I can't see anywhere that we use the slug where changing it would cause issues, eg. the patterns only store the `id` and the mapping to `slugs` via the `name` is all done on the fly, but I don't think we necessarily need to complicate things at this point by generating new slugs as they are not visible to users anywhere. If renaming a category via `/wp-admin/term.php?taxonomy=wp_pattern_category` the slug remains the same unless you explicitly change it so it seems ok to follow the same approach here I think.

Reply:
Thanks for the feedback 👍 

I've added the normalization of the category object to the rename menu item component for the time being. 

As for the slug, I agree. If we do in future decide to go down that path, it should be a straightforward follow-up.